src/Altinn.App.PlatformServices/Implementation/ApplicationLanguage.cs

```diff
@@ -0,0 +1,90 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text.Json;
+using System.Threading.Tasks;
+using Altinn.App.PlatformServices.Helpers;
+using Altinn.App.PlatformServices.Interface;
+using Altinn.App.Services.Configuration;
+using Altinn.App.Services.Implementation;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+
+namespace Altinn.App.PlatformServices.Implementation
+{
+    /// <summary>
+    /// An implementation used to retrieve the supported application languages.
+    /// </summary>
+    public class ApplicationLanguage : IApplicationLanguage
+    {
+        private readonly AppSettings _settings;
+        private readonly ILogger _logger;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ApplicationLanguage"/> class.
+        /// </summary>
+        /// <param name="settings">The app repository settings.</param>
+        /// <param name="logger">A logger from the built in logger factory.</param>
+        public ApplicationLanguage(
+            IOptions<AppSettings> settings,
+            ILogger<AppResourcesSI> logger)
+        {
+            _settings = settings.Value;
+            _logger = logger;
+        }
+
+        /// <inheritdoc />
+        public async Task<List<Platform.Storage.Interface.Models.ApplicationLanguage>> GetApplicationLanguages()
+        {
+            string pathTextsResourceFolder = _settings.AppBasePath + _settings.ConfigurationFolder + _settings.TextFolder;
+
+            DirectoryInfo directoryInfo = new DirectoryInfo(pathTextsResourceFolder);
+
+            if (!directoryInfo.Exists)
+            {
+                _logger.LogWarning("The text resource directory does not exist");
+                return new List<Platform.Storage.Interface.Models.ApplicationLanguage>();
+            }
+
+            if (directoryInfo.GetFiles().Length < 1)
```

Comment:
Not really any need for this as your main foreach should return an empty list if there are no files. Ie. create an empty list before the foreach, add any languages you find, if no files are found just return the empty list. You can choose to log a warning in the case of no resource files found, but the empty list in itself would be enough to identify the issue.